src/api/worker/crypto/InstanceMapper.ts

```diff
@@ -29,6 +29,9 @@ export class InstanceMapper {
 	 * @returns The decrypted and mapped instance
 	 */
 	decryptAndMapToInstance<T>(model: TypeModel, instance: Record<string, any>, sk: Aes128Key | null): Promise<T> {
+		if (model.encrypted && sk == null) {
```

Comment:
would it make sense to ensure this statically with `TypeModel & {encrypted: true}` and remove the `null`  from sk? Also, what's the relationship between TypeModel.encrypted and ModelValue.encrypted?

Reply:
Not really. We call it in a bunch of places where we don't know anything static about the instance. Nor do we have a bespoke type for each `TypeModel` which would let us enforce it. We would have to assert it in each case and I don't know if it's necessary.